### PR TITLE
(feat) Export OpenmrsDatePickerProps interface to facilitate testing of the OpenmrsDatePicker

### PR DESCRIPTION
**Pull request by kajambiya**
# Requirements

- [ ] This PR has a title that briefly describes the work done including the ticket number. Ensure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing#contributing-guidelines) label (such as `feat`, `fix`, or `chore`, among others). See existing PR titles for inspiration.

## For changes to apps

- [ ] My work conforms to the [**O3 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://om.rs/o3ui).

## If applicable

- [ ] My work includes tests or is validated by existing tests.
- [ ] I have updated the [esm-framework mock](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-framework/mock.tsx) to reflect any API changes I have made.

## Summary
When writing tests for the OpenmrsDatePicker, jest was failing to recognize the component because the OpenmrsDatePickerProps interface was not being exported. This PR fixes the issue.

### CODE REVIEW

**Comment by arodidev** [comment]
@kajambiya is there a way we can add any visual representation to better understand the effects of the changes made?

File: packages/framework/esm-framework/mock.tsx [hunk 1]
```diff
@@ -58,7 +58,6 @@ export const LeftNavMenu = jest.fn(() => <div>Left Nav Menu</div>);
 export const setLeftNav = jest.fn();
 export const unsetLeftNav = jest.fn();
 export const ResponsiveWrapper = jest.fn(({ children }) => <>{children}</>);
-export const OpenmrsDatePicker = jest.fn(() => <div>OpenMRS DatePicker</div>);
```
Comment (ibacher) [inline packages/framework/esm-framework/mock.tsx:61]: Why was this removed from the mock?
